### ISSUE TITLE
As a user, I want to change extended options on my service instance

### ISSUE THREAD

**Issue by afeld** [18F/cf-cdn-service-broker#50]
The intention of https://github.com/18F/cf-cdn-service-broker/issues/43 was to support changing the two options (currently) accepted through creation of the service instance. In https://github.com/18F/cf-cdn-service-broker/issues/43#issuecomment-268934628, @amaline requested changing an option that just gets a default value right now. I see a couple of ways we could tackle this:

* Add options to the broker, such as `forward_cookie`
* Enable the broker to create service keys with an associated IAM policy, so that users can modify the CloudFront instances themselves directly

The latter is the approach we're taking with the S3 broker and are planning to take with the RDS broker, so I suggest we go with that.

**Comment by amaline** [comment]
I would agree with option two.  We just identified another use case we need which is updating the CloudFront distribution to forward access logs<http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html> to an S3 bucket.

Thanks,

Al

**Comment by mogul** [comment]
> Enable the broker to create service keys with an associated IAM policy, so that users can modify the CloudFront instances themselves directly

This is looking like the general pattern that makes sense... Expose the relevant AWS service backing with a limited IAM role. 

I've always been against exposing too much of the IaaS provider-specifics to avoid dependencies on that particular provider (both for cloud.gov to provide these services and for our customers in using them), but it does seem to be the simplest way to expose functionality that we'd otherwise just be wrapping in the broker, when we could/should be putting that effort into other features. And knowing that people like @amaline are stepping up and saying "that's what I want" (eg it's not too much to ask that people use the AWS API/CLI to make "advanced" configuration) definitely helps, so thanks!

**Comment by amaline** [comment]
Thanks for the callout @mogul.  I do agree with you that it would be ideal to have service brokers that completely isolated the end user from the IaaS.  Especially when desiring the ability to recover from a complete technology outage at one cloud service provider (which is also why the S3 service broker is poorly named BTW, BlobStore would be better).  However, until the a service broker is feature complete there will be a trade-off between getting real workloads up and running avoiding infrastructure dependencies.

**Comment by mogul** [comment]
I agree on renaming S3; we'll look into it. (I had the same concern about RDS, which is opaque to people not familiar with AWS; will likely rename to `RDBMS` at some point.)

I'm inclined to go this route for a while and see what IAM permissions we actually end up having to grant in response to user requests. Then at a future date when we have the leisure (ha!) we can expose the most common configurations via broker parameters and reduce reliance on this exposure.

**Comment by jmcarp** [comment]
This might not be possible at the moment: http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_aws-services-that-work-with-iam.html

More details at http://stackoverflow.com/questions/29558655/restrict-access-to-a-particular-cloudfront-distribution-using-iam.
